core-web/src/main/java/org/silverpeas/core/web/treemenu/process/TreeBuilder.java

```diff
@@ -83,7 +82,6 @@ public static MenuItem buildLevelMenu(TreeFilter filter, MenuItem father, String
    * @param userId the user identifier to display only the authorized item menu
    * @param language the user language to display the label menu in the correct language
    * @return a level of the menu
-   * @throws RemoteException
    */
   private static MenuItem buildOtherLevel(TreeFilter filter, MenuItem father, String userId,
```

Comment:
This method is far complex: too long with a lot of if-statements. SonarCloud agrees with me. Please, split its body in several little private methods.

Reply:
Done.